src/QMCWaveFunctions/SPOSet.h

```diff
@@ -256,6 +256,18 @@ class SPOSet: public QMCTraits
   evaluate(const ParticleSet& P, int iat,
            ValueVector_t& psi, GradVector_t& dpsi, HessVector_t& grad_grad_psi)=0;
 
+  /** evaluate the values, gradients, hessians, and grad hessians of this single-particle orbital set
+   * @param P current ParticleSet
+   * @param iat active particle
+   * @param psi values of the SPO
+   * @param dpsi gradients of the SPO
+   * @param grad_grad_psi hessians of the SPO
+   * @param grad_grad_grad_psi grad hessians of the SPO
+   */
+  virtual void
+  evaluate(const ParticleSet& P, int iat,
+           ValueVector_t& psi, GradVector_t& dpsi, HessVector_t& grad_grad_psi, GGGVector_t& grad_grad_grad_psi)=0;
```

Comment:
I think it is better to define APP_ABORT as the base class implementation instead of implementing every APP_ABORT at every place.

Reply:
Sounds reasonable.  I'll do that instead.

Reply:
Could you also take care of the above function? It think it has the same situation.